services/QuillLMS/app/workers/batch_save_activity_session_concept_results_worker.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+class BatchSaveActivitySessionConceptResultsWorker
+  include Sidekiq::Worker
+
+  sidekiq_options queue: SidekiqQueue::CRITICAL
+
+  def perform(concept_results, activity_session_id, activity_session_uid)
+    batch_runner(activity_session_uid) do
+      concept_results
+        .map { |cr| concept_results_hash(cr, activity_session_id) }
+        .reject(&:empty?)
+        .each { |crh| SaveActivitySessionConceptResultsWorker.perform_async(crh) }
+    end
+  end
+
+  def on_complete(status, options)
+    if status.failures == 0
+      PusherTrigger.run(options['activity_session_uid'], 'concept-results-saved', "Concept results saved for activity session with uid: #{options['activity_session_uid']}")
+    else
+      PusherTrigger.run(options['activity_session_uid'], 'concept-results-partially-saved', "Concept results partially saved for activity session with uid: #{options['activity_session_uid']}, #{status.total - status.failures} succeeded, #{status.failures} failed")
```

Comment:
We might want to add some logging here for failures, such as a ChangeLog (e.g. see [recalculate_locks](https://github.com/empirical-org/Empirical-Core/blob/develop/services/QuillLMS/app/models/change_log.rb#L104)).  Otherwise the student gets redirected but we have no idea how much this might be happening.

Reply:
We'll still see it [here](https://quillorg-5s.sentry.io/issues/?groupStatsPeriod=auto&page=0&project=11238&query=is%3Aunresolved+saveactivitysessionconceptresults&referrer=issue-list&statsPeriod=14d), though, right? I am personally inclined to feel like our existing error tracking is enough if we want to actually deal with the root cause here, but that seems like a separate concern to me.

Reply:
Good point.   Carry on!